passport_strategies/initLocalAuth.js

```diff
@@ -9,21 +9,25 @@ module.exports = function localAuthStrategy(passport) {
     User.findByEmail(email).then((user) => {
       if (!user) {
         console.log(`Email does not exist: ${email}`);
-        return done(null, false, {
+        done(null, false, {
           message: 'bad email',
         });
-      }
-      if (password !== user.password) {
-        console.log(`Incorrect password for user: ${user.email}`);
-        return done(null, false, {
-          message: 'bad password',
+      } else {
+        User.comparePassword(password, user.password).then((success) => {
+          if (success) {
+            console.log(`Successfully Authenticated: ${user.email}`);
+            done(null, user);
+          } else {
+            console.log(`Incorrect password for user: ${user.email}`);
```

Comment:
We should send the same message: "incorrect username or password" for both !user and !success.
We don't want the user to know which field they entered incorrectly. The front end will directly load the message from json body.